drivers/spi/spi_context.h

```diff
@@ -144,36 +146,40 @@ static inline void spi_context_cs_control(struct spi_context *ctx, bool on)
 }
 
 static inline void spi_context_buffers_setup(struct spi_context *ctx,
-					     const struct spi_buf **tx_bufs,
-					     struct spi_buf **rx_bufs,
+					     const struct spi_buf *tx_bufs,
+					     size_t tx_count,
+					     struct spi_buf *rx_bufs,
+					     size_t rx_count,
 					     uint8_t dfs)
 {
-	SYS_LOG_DBG("tx_bufs %p (%p) - rx_bufs %p (%p) - %u",
-		    tx_bufs, tx_bufs ? *tx_bufs : NULL,
-		    rx_bufs, rx_bufs ? *rx_bufs : NULL, dfs);
+	SYS_LOG_DBG("tx_bufs %p (%d) - rx_bufs %p (%d) - %u",
+		    tx_bufs, tx_count, rx_bufs, rx_count, dfs);
```

Comment:
You'll likely get compiler warnings from this for incorrect format specifiers. The right one for size_t is %zu.